### PR TITLE
Use async/await to properly simulate long network I/O

### PR DESCRIPTION
**Pull request by AndrewJo**

### CODE REVIEW

File: javascript/batched_queries.js [hunk 1]
```diff
@@ -88,14 +80,18 @@ const queries = [
...
-let end = Date.now();
-console.log(`Time taken: ${end - start} ms`);
-
-console.log('== SOLUTION ==');
-start = Date.now();
-runAllQueriesBatched(queries);
-end = Date.now();
-console.log(`Time taken: ${end - start} ms`);
-
+(async function () {
```
Comment (wwc278) [inline javascript/batched_queries.js]: Is it possible to use arrow functions here?

File: javascript/batched_queries.js [hunk 2]
```diff
@@ -2,13 +2,6 @@
...
 'use strict';
 
-Number.prototype.toFixedDown = function (digits) {
-  /*eslint prefer-template: "off"*/
-  const re = new RegExp('(\\d+\\.\\d{' + digits + '})(\\d)');
-  const m = this.toString().match(re);
-  return m ? parseFloat(m[1]) : this.valueOf();
-};
-
 String.prototype.numberify = function (timestamp) {
```
Comment (wwc278) [inline javascript/batched_queries.js]: Is it possible to use arrow functions here?

**Review by wwc278 — CHANGES_REQUESTED** [review]
I found traces of old `function` syntax and would like to standardize this file to arrow functions as much as possible. Everything else looks good.

**Comment by AndrewJo** [comment]
@wwc278 Made further changes:

- Changed to arrow function notations
- Create an async sleep function so the code reads more naturally

**Review by wwc278 — APPROVED** [review]

**Pull request merged**